lib/client.js

```diff
@@ -467,7 +467,14 @@ function Request(method, url) {
       self.emit('response', res);
     }
 
-    self.callback(err, res);
+    if (res && res.status >= 200 && res.status < 300) {
```

Comment:
They are errors for ajax requests because they don't have bodies themselves or tell you much about the success of the request.